Cabal/doc/developing-packages.rst

```diff
@@ -2162,7 +2162,8 @@ Configuration Flags
 
    Flag section declares a flag which can be used in `conditional blocks`_.
 
-A flag section may contain the following fields:
+Flag names are case-insensitive and must match ``[a-z0-9_][a-z0-9_-]*``
```

Comment:
That limitation only applies to Hackage as a point of distribution.

Here's a perfectly valid `.cabal` file including unicode package names and flag names:

```
name:                無
version:             0
synopsis:            The canonical non-package
build-type:          Simple
cabal-version:       >=1.10

source-repository head
  Type:     git
  Location: https://github.com/hvr/-.git

flag 無
  description: 無

library
  default-language:    Haskell2010

  if !flag(無)
    buildable:False
```